### PR TITLE
Da 2812 write dockerhub instrumentation in golang

### PR DESCRIPTION
**Pull request by enkhalifapro**

### CODE REVIEW

**Comment by lukaszgryglicki** [comment]
Is this ready for review? We should merge this together with `da-ds` PR, right?

File: data/onap.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+native:
+  slug: lfn/onap
+data_sources:
+- slug: dockerhub
+  endpoints:
+  - name: onap
+    flags:
+      type: dockerhub_org
```
Comment (lukaszgryglicki) [inline data/onap.yaml:9]: Please don't add files to this directory, SDS assumes it is empty and clones all our fixtures there. If you need examples add them to "examples" directory.

File: sources/cmd/syncdatasources/syncdatasources.go [hunk 2]
```diff
@@ -100,6 +100,13 @@ var (
 			"DA_ROCKETCHAT_NO_SSL_VERIFY": "1",
 			"DA_ROCKETCHAT_WAIT_RATE":     "1",
 		},
+		lib.DockerHub: {
+			"DA_DOCKERHUB_USERNAME": "",
```
Comment (lukaszgryglicki) [inline sources/cmd/syncdatasources/syncdatasources.go:104]: Why do we need to add a default value that is empty/unusable? Those defaults are only needed when they set some useful value.

File: sources/context.go [hunk 3]
```diff
@@ -131,7 +131,7 @@ func (ctx *Ctx) Init() {
 	// ElasticSearch
 	ctx.ElasticURL = os.Getenv("SDS_ES_URL")
 	if ctx.ElasticURL == "" {
-		ctx.ElasticURL = "http://127.0.0.1:9200"
+		ctx.ElasticURL = "http://elastic:changeme@127.0.0.1:9200"
```
Comment (lukaszgryglicki) [inline sources/context.go:134]: Why this change? This will break OOTB working with local elastic search instance running from docker...

File: sources/go.mod [hunk 4]
```diff
@@ -0,0 +1,12 @@
+module github.com/LF-Engineering/sync-data-sources/sources
+
+go 1.13
```
Comment (lukaszgryglicki) [inline sources/go.mod:3]: Why go 1.13 while the current one is 1.15? cc @1010sachin @fayazg if we start to lock into a specific go version, let's lock with 1.15 - I'm using 1.15 locally and for SDS builds for a long time and it works - current SDS running on our EC2 prod/test also uses 1.15, why switch to an older version (and why even specify it)?

Comment (fayazg) [reply sources/go.mod:3]: @lukaszgryglicki we need to start building against specific versions since not everyone working on this might be using the same version.
@enkhalifapro is there a conflict between 1.15 and anything you have added?

Comment (lukaszgryglicki) [reply sources/go.mod:3]: So let's use the newest, I'm still thinking this only matters for deployments (like fix the version to `v1.15` in CI deployment or while building the final image, we should not have such restrictions for local setup IMHO).

**Comment by lukaszgryglicki** [comment]
When I `git checkout` this branch, `make` fails:
```
2020-12-02 07:18:52 root@teststats:~/dev/go/src/github.com/LF-Engineering/sync-data-sources/sources# make
./for_each_go_file.sh "gofmt -s -w"
./for_each_go_file.sh "golint -set_exit_status"
./for_each_go_file.sh "goimports -w"
./vet_files.sh "go vet"
go: downloading golang.org/x/oauth2 v0.0.0-20201109201403-9fd604954f58
go: downloading github.com/google/go-github v17.0.0+incompatible
go: downloading golang.org/x/net v0.0.0-20200822124328-c89045814202
go: downloading github.com/kr/pretty v0.1.0
# command-line-arguments
cmd/syncdatasources/syncdatasources.go:4325:14: Fatalf call has arguments but no formatting directives
Makefile:41: recipe for target 'vet' failed
make: *** [vet] Error 1
```
When I attempt to `git checkout` master later, git complains about unsaved changes:
```
2020-12-02 07:19:22 root@teststats:~/dev/go/src/github.com/LF-Engineering/sync-data-sources/sources# git checkout master
error: Your local changes to the following files would be overwritten by checkout:
	sources/cmd/syncdatasources/syncdatasources.go
	sources/go.sum
Please commit your changes or stash them before you switch branches.
Aborting
```
This is because vet/lint/fmt changes are not saved:
Screenshot 2020-12-02 at 08 21 03

Finally when I `git reset --hard`, `checkout master` `make` works correctly:
```
2020-12-02 07:22:02 root@teststats:~/dev/go/src/github.com/LF-Engineering/sync-data-sources/sources# git reset --hard
HEAD is now at 3d32f26 DA-2812: Add project to repo
2020-12-02 07:22:10 root@teststats:~/dev/go/src/github.com/LF-Engineering/sync-data-sources/sources# git checkout master
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
2020-12-02 07:22:14 root@teststats:~/dev/go/src/github.com/LF-Engineering/sync-data-sources/sources# make
./for_each_go_file.sh "gofmt -s -w"
./for_each_go_file.sh "golint -set_exit_status"
./for_each_go_file.sh "goimports -w"
./vet_files.sh "go vet"
goconst ./...
usedexports ./...
errcheck -asserts -ignore '[FS]?[Pp]rint*' -ignoretests ./...
CGO_ENABLED=0 go build -ldflags '-s -w' -o syncdatasources cmd/syncdatasources/syncdatasources.go
CGO_ENABLED=0 go build -ldflags '-s -w' -o sds-crontab cmd/sds-crontab/sds-crontab.go
CGO_ENABLED=0 go build -ldflags '-s -w' -o gen-regexp cmd/gen-regexp/gen-regexp.go
2020-12-02 07:22:23 root@teststats:~/dev/go/src/github.com/LF-Engineering/sync-data-sources/sources# 
```
Please address.
